components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SAML2SSOManager.java

```diff
@@ -365,6 +368,77 @@ public void processResponse(HttpServletRequest request, HttpServletResponse resp
         }
     }
 
+    /**
+     * Process authentication response with SAML2 artifact.
+     *
+     * @param request Http Servlet Request object.
+     * @throws SSOAgentException
+     */
+    public void processArtifactResponse(HttpServletRequest request) throws SSOAgentException {
+
+        SAMLSSOArtifactResolutionService artifactResolutionService =
+                new SAMLSSOArtifactResolutionService(ssoAgentConfig);
+        try {
+            ArtifactResponse artifactResponse = artifactResolutionService.getSAMLArtifactResponse(
+                    request.getParameter(SSOAgentConstants.SAML2SSO.SAML2_ARTIFACT_RESP));
+
+            if (!StringUtils.equals(artifactResponse.getStatus().getStatusCode().getValue(), StatusCode.SUCCESS_URI)) {
+                throw new SSOAgentException("Received an invalid SAML response.");
+            }
+            if (artifactResponse.getMessage() == null) {
+                throw new SSOAgentException("Received SAML2 Artifact response message was null.");
+            }
+            XMLObject xmlObject = artifactResponse.getMessage();
+            if (xmlObject instanceof Response || xmlObject instanceof LogoutResponse) {
```

Comment:
We can't silently return in a case if this condition is not met

Reply:
fixed.